tests/infrastructure/src/main/java/org/mule/test/infrastructure/process/Controller.java

```diff
@@ -237,6 +237,10 @@ protected boolean isDeployed(String appName) {
     return new File(appsDir, appName + ANCHOR_SUFFIX).exists();
   }
 
+  protected boolean isUndeployed(String appName) {
+    return !(new File(appsDir, appName + ANCHOR_SUFFIX).exists() || new File(appsDir, appName).exists());
```

Comment:
If an application folder exists, but the anchor file doesn't, this would return false, which is not true

Reply:
🤔 That's precisely the intention of the method. 
My idea, from the lesson, learned of the classloading test case, is that if there's a folder with the same name as the app, that means that the mechanism of un-deployment didn't work well and the app wasn't undeployed successfully. So I want the user to know that the app isn't fully un-deployed.